### PR TITLE
STRF-9610: node 14 support + npm 7/8, drop < 12

### PR DESCRIPTION
**Pull request by jairo-bc**

### CODE REVIEW

**Comment by jairo-bc** [comment]
Regarding failing windows test, it is exiting in windows for some reason with an error.
Other environments(linux and macos) and previously in master, they have same result as windows, but not exiting with an error.
Screenshot 2022-01-28 at 20 04 36

**Comment by zvuki** [comment]
@jairo-bc what about `build-darwin-node-14`?

**Comment by jairo-bc** [comment]
@zvuki restarted it and now looks good.

File: .circleci/config.yml [hunk 1]
```diff
@@ -51,115 +55,102 @@ commands:
             echo 'export NVM_DIR="$HOME/.nvm"' >> $BASH_ENV
             echo "[ -s \"$NVM_DIR/nvm.sh\" ] && . \"$NVM_DIR/nvm.sh\"" >> $BASH_ENV
 jobs:
-  build-linux-node-6:
+  build-linux-node-10:
```
Comment (mattolson) [inline .circleci/config.yml:58]: I don't think we need Node 10 or 11 anymore since it is now EOL.

File: .circleci/config.yml [hunk 2]
```diff
@@ -51,115 +55,102 @@ commands:
...
+      - image: circleci/node:12
     steps:
       - build
       - release
 
-  build-linux-node-9:
+  build-linux-node-13:
     docker:
-      - image: circleci/node:9
+      - image: circleci/node:13
```
Comment (mattolson) [inline .circleci/config.yml:81]: I don't think we need Node 13. Let's just stick with the LTS versions -- 12, 14, 16

File: package.json [hunk 3]
```diff
@@ -15,7 +15,7 @@
     "url": "http://andrew.github.com"
   },
   "engines": {
-    "node": ">= 6.0.0 <13.0.0"
+    "node": ">= 10.0.0 <15.0.0"
```
Comment (mattolson) [inline package.json:18]: Let's bump the min to 12

File: test/spec.js [hunk 4]
```diff
@@ -38,22 +38,25 @@ describe('spec', function() {
...
-              if (expected) {
-                assert.equal(util.normalize(result.css.toString()), expected);
-              }
-              done();
-            });
+            // var expected = util.normalize(read(t.expected, 'utf8'));
+            assert(true);
+            done();
+            return;
+            // sass.render({
```
Comment (mattolson) [inline test/spec.js:45]: Why do we need to remove this test?

Comment (jairo-bc) [reply test/spec.js:45]: sorry, I'm testing the issue with the windows, when process doesn't exist when running `npm test`

File: .circleci/config.yml (hunk 2 above)
Comment (jairo-bc) [reply .circleci/config.yml:81]: I left those job definitions, but they are not being run. 
https://github.com/bigcommerce-labs/node-sass/pull/33/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L247

File: appveyor.yml [hunk 5]
```diff
@@ -55,6 +55,12 @@
       - nodejs_version: 12
         GYP_MSVS_VERSION: 2017
         APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2017
+      - nodejs_version: 13
```
Comment (mattolson) [inline appveyor.yml:58]: I don't think you need 13 anymore now that 14 is out. This will just slow down the build for no real benefit.

Comment (jairo-bc) [reply appveyor.yml:58]: yeah, basically we don't need any changes in this file. We are not using appveyor for deployments.

**Review by zvuki — APPROVED** [review]

**Pull request merged**